model/Service/NavigationContextAwareTrait.php

```diff
@@ -35,11 +35,11 @@ trait NavigationContextAwareTrait
     /** @var string|null*/
     private $ref = null;
 
-    public function setNavigationContext(string $direction, string $scope, ?string $ref): void
+    public function setNavigationContext(?string $direction, ?string $scope, ?string $ref): void
     {
-        $this->direction = $direction;
-        $this->scope = $scope;
-        $this->ref = $ref;
+        $this->direction = $direction ?? '';
+        $this->scope = $scope ?? '';
+        $this->ref = $ref ?? null;
```

Comment:
I found a bit misleading that we allow null, but only `ref` receives null and only `getRef() ?string` has return type null. I think the fix is solving the issue, but may also be causing some integrity issues. We will never know whatever user passed empty or null, might be that for empty we wanted one behavior and for null another, now it will be difficult to check this.

Btw, you do not need this check for $ref.

```suggestion
        $this->direction = $direction ?? '';
        $this->scope = $scope ?? '';
        $this->ref = $ref;
```

Reply:
I've applied your change in 7b78cf9690c5c5d116bcf2663b12301489de5ab2 to keep `$ref` as provided in the ctor, thanks!

Right now I'm setting `$scope` and `$direction` to empty strings because if we don't I think we'd need to change the return type for getters in this trait, and I don't want to change that many places in this PR :sweat_smile: